aiozk/recipes/base_lock.py

```diff
@@ -15,7 +15,18 @@ class BaseLock(SequentialRecipe):
         if timeout is not None:
             time_limit = time.time() + timeout
 
-        await self.create_unique_znode(znode_label)
+        try:
+            await self.create_unique_znode(znode_label)
+        except Exception as ex:
+            log.debug('unknown exception: %s', ex)
+            while True:
+                try:
+                    await self.delete_garbage_znodes(znode_label)
+                except Exception as e:
+                    log.debug('unknown exception: %s', e)
+                else:
+                    break
+            raise ex
```

Comment:
Probably all this code should be moved inside of `create_unique_znode` because it affects all other cases in similar way too: election, party, lease and double barrier currently are using it without handling error cases.
So in the case of an error when we're doing `create_unique_znode` we will cleanup nodes that were created and not consumed.

Reply:
Ok I will try.